Sally7/S7Connection.cs

```diff
@@ -100,8 +101,7 @@ public async Task OpenAsync()
             await TcpClient.ConnectAsync(host, IsoOverTcpPort).ConfigureAwait(false);
             var stream = TcpClient.GetStream();
 
-            // TODO: use memory from the pool
-            var buffer = new byte[100];
+            var buffer = ArrayPool<byte>.Shared.Rent(100);
```

Comment:
This will actually rent an array of size rounded up to the next power of two, so the length will be 128.
This shouldn't be problem, as the implementation in the S7Connection is defensive for that.